llvm/lib/Target/AIE/AIEBaseAliasAnalysis.cpp

```diff
@@ -117,13 +135,170 @@ static const Value *getUnderlyingObjectAIE(const Value *V) {
   return V;
 }
 
+hash_code hashIntrinsic(const Value *V) {
+  const auto *Extract = dyn_cast<ExtractValueInst>(V);
+  assert(Extract && "Not an Extract");
+  const auto *IntrinsicCall =
+      dyn_cast<CallBase>(Extract->getAggregateOperand());
+  assert(IntrinsicCall && "Not an Intrinsic");
+  Intrinsic::ID ID = IntrinsicCall->getIntrinsicID();
+  // Intrinsic ID is part of the hash.
+  hash_code Hash = hash_value(ID);
+  unsigned InPtrIdx;
+  isAIEPtrAddIntrinsic(ID, InPtrIdx);
+  // Necessary operands to desambiguate.
+  for (unsigned Op = InPtrIdx + 1; Op < hashableAddIntrinsicOps(ID); Op++) {
+    Hash = hash_combine(Hash, hash_value(IntrinsicCall->getArgOperand(Op)));
+  }
+  return Hash;
+}
+
+static std::pair<int, short> trackIntrinsicChain(const Value *V) {
```

Comment:
Could you add some more comment here? Especially what the returned tuple represents.